src/Util.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace React\Http;
+
+/**
+ * @internal
+ */
+final class Util
+{
+    /**
+     * Convert a ini like size to a numeric size in bytes.
+     *
+     * @param string $iniSetting
+     * @return int
+     */
+    public static function iniSizeToBytes($size)
+    {
+        if (is_numeric($size) && $size <= 0) {
+            throw new \InvalidArgumentException('Expect "' . $size . '" to be higher isn\'t zero or lower');
+        }
```

Comment:
What is the motivation for this check? `memory_limit` can be set to `-1` to allow unlimited memory (http://php.net/manual/en/ini.core.php#ini.memory-limit), while other ini settings do not allow this.

Should this be covered by this method?

Reply:
Good point, motivation for it was to catch invalid values but it escaped my mind that -1 is a valid value